sample/kotlin/src/main/kotlin/com/datadog/android/sample/profiling/HeavyCPUWork.kt

```diff
@@ -0,0 +1,199 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sample.profiling
+
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.math.cos
+import kotlin.math.pow
+import kotlin.math.sin
+import kotlin.math.sqrt
+import kotlin.random.Random
+
+/**
+ * Simulates heavy CPU computation for profiling purposes.
+ * This class performs various CPU-intensive operations including:
+ * - Complex mathematical calculations
+ * - String manipulation
+ * - List/array operations
+ * - Prime number calculations
+ */
+@Suppress("MagicNumber", "TooManyFunctions")
+class HeavyCPUWork {
+
+    private val isRunning = AtomicBoolean(false)
+    private var workerThread: Thread? = null
+
+    /**
+     * Starts the heavy CPU computation on a background thread.
+     */
+    fun start() {
+        if (isRunning.getAndSet(true)) {
+            // Already running
+            return
+        }
+
+        workerThread = Thread {
+            while (isRunning.get()) {
+                performHeavyComputation()
+            }
+        }.apply {
+            name = "HeavyCPUWork-Thread"
+            start()
+        }
+    }
+
+    /**
+     * Stops the heavy CPU computation.
+     */
+    fun stop() {
+        isRunning.set(false)
+        workerThread?.interrupt()
```

Comment:
I don't think it is needed, since there is nothing in the thread workload what consumes this flag (no locks, monitors or is interrupted check). Setting `isRunning.set(false)` is already enough to stop the run.